frontend/src/component/PublicProfile/CustomFormManager.jsx

```diff
@@ -206,28 +207,46 @@ const TextRichWidget = connect(({ user }) => ({ user }))(props => {
   )
 })
 
-const SchemaAsForm = props => {
-  return (
-    <Form
-      schema={props.schemaObject}
-      uiSchema={props.uiSchemaObject}
-      formData={props.dataSchemaObject}
-      widgets={{
-        TextareaWidget: TextRichWidget
-      }}
-      onSubmit={(formData, e) => {
-        props.onClickToggleButton()
-        props.onSubmitDataSchema(formData, e)
-      }}
-    >
-      <IconButton
-        customClass={props.submitButtonClass}
-        icon='check'
-        type='submit'
-        text={props.validateLabel}
-      />
-    </Form>
-  )
+class SchemaAsForm extends React.Component {
```

Comment:
Shouldn't this be in a different jsx file ?

Reply:
this file holds 6 components
each of them could be in a specific file but I chose not to to save development time

once we will refactor it to make it more generic with the app custom forms, we will can split it

Reply:
ok, works for me